gcc/rust/typecheck/rust-hir-type-check-pattern.cc

```diff
@@ -204,7 +204,7 @@ TypeCheckPattern::visit (HIR::StructPattern &pattern)
 	    if (!variant->lookup_field (ident.get_identifier (), &field,
 					nullptr))
 	      {
-		rust_error_at (ident.get_locus (),
+		rust_error_at (ident.get_locus (), ErrorCode ("E0026"),
```

Comment:
Both the lines you modified have the exact same error message and pattern. I think it would be worth opening an issue to factor them into one function (`emit_invalid_field_error` or w/ever) in this file, and it's something you can work on if you like. This will make it easier for you to rework the Error emitting code once we get to that point in your GSoC project @MahadMuhammad